lib/rules/no-implicit-coercion.js

```diff
@@ -299,6 +304,33 @@ module.exports = {
 
                     report(node, recommendation, true);
                 }
+            },
+
+            TemplateLiteral(node) {
+                if (!options.templateString) {
+                    return;
+                }
+
+                // `` or `${foo}${bar}`
+                if (node.expressions.length !== 1) {
+                    return;
+                }
+
+
+                //  `prefix${foo}`
+                if (node.quasis[0].value.raw !== "") {
+                    return;
+                }
+
+                //  `${foo}postfix`
+                if (node.quasis[1].value.raw !== "") {
+                    return;
+                }
```

Comment:
Should we check `raw` or `cooked`? For example, if we're checking `raw`, then the following would be allowed:

```js
`\
${foo}`;

`${foo}\
`;
```

Reply:
I defnitely think checking `cooked` is better.